### PR TITLE
RUM-4911 feat(watchdog-termination): setup trackWatchdogTermination configuration

### PR DESCRIPTION
**Pull request by ganeshnj**
### What and why?

Ability to enable disable WT. Default is false.

### How?

We don't want to enable WT by default as they will have huge disk IO implications.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [ ] Run integration tests
- [ ] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

File: DatadogRUM/Sources/RUMConfiguration.swift [hunk 1]
```diff
@@ -111,6 +111,13 @@ extension RUM {
         /// Default: `false`.
         public var trackBackgroundEvents: Bool
 
+        /// Determines whether the SDK should track application termination by the watchdog.
+        ///
+        /// Read more about watchdog terminations at https://developer.apple.com/documentation/xcode/addressing-watchdog-terminations
+        ///
+        /// Default: `false`.
+        public var trackWatchdogTermination: Bool
```
Comment (ncreated) [inline DatadogRUM/Sources/RUMConfiguration.swift:119]: convention/ Other `track*` options names are plural:
- trackBackgroundEvent**s**
- trackFrustration**s**

For consistency this one could be too:
```diff
- trackWatchdogTermination
+ trackWatchdogTerminations
```

File: DatadogRUM/Sources/Instrumentation/WatchdogTerminations/WatchdogTerminationMonitor.swift [hunk 2]
```diff
@@ -45,7 +45,7 @@ internal final class WatchdogTerminationMonitor {
             try appStateManager.start()
         } catch let error {
             DD.logger.error(ErrorMessages.failedToStartAppState, error: error)
-            telemetry.error(ErrorMessages.failedToStartAppState, error: error)
+            feature.telemetry.error(ErrorMessages.failedToStartAppState, error: error)
```
Comment (ncreated) [inline DatadogRUM/Sources/Instrumentation/WatchdogTerminations/WatchdogTerminationMonitor.swift:48]: nice 💪

**Review by ncreated — DISMISSED** [review]
Looks good 👍 with two non-blocking remarks:

- I left a comment on the option name,
- It needs to be also available for objc:

https://github.com/DataDog/dd-sdk-ios/blob/b6327d1b13e758861cc53e917a083a5b545fe331/DatadogObjc/Sources/RUM/RUM%2Bobjc.swift#L327-L328

File: DatadogRUM/Sources/RUMConfiguration.swift (hunk 1 above)
Comment (ganeshnj) [reply DatadogRUM/Sources/RUMConfiguration.swift:119]: sure.

**Review by ncreated — APPROVED** [review]

**Pull request merged**